challenges-services/src/main/java/org/exoplatform/challenges/listener/NewAnnouncementActivityGenerator.java

```diff
@@ -0,0 +1,81 @@
+package org.exoplatform.challenges.listener;
+
+import org.apache.commons.lang.StringEscapeUtils;
+import org.exoplatform.challenges.model.Announcement;
+import org.exoplatform.challenges.model.UserInfo;
+import org.exoplatform.challenges.storage.AnnouncementStorage;
+import org.exoplatform.challenges.utils.Utils;
+import org.exoplatform.commons.exception.ObjectNotFoundException;
+import org.exoplatform.services.listener.Event;
+import org.exoplatform.services.listener.Listener;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.social.core.activity.model.ExoSocialActivity;
+import org.exoplatform.social.core.activity.model.ExoSocialActivityImpl;
+import org.exoplatform.social.core.identity.model.Identity;
+import org.exoplatform.social.core.space.model.Space;
+import org.exoplatform.social.core.storage.api.ActivityStorage;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class NewAnnouncementActivityGenerator extends Listener<AnnouncementStorage, Announcement> {
+  private static final Log LOG = ExoLogger.getLogger(NewAnnouncementActivityGenerator.class);
+
+  public static final String ANNOUNCEMENT_ACTIVITY_TYPE = "exo-announcement:activity";
```

Comment:
challenges-announcement
```suggestion
  public static final String ANNOUNCEMENT_ACTIVITY_TYPE = "challenges-announcement";
```